cookbook/case_studies/house_price_prediction/multiregion_house_price_predictor.py

```diff
@@ -0,0 +1,158 @@
+"""
+Predicting House Prices in Multiple Regions Using an XGBoost Model and Flytekit (Python)
+----------------------------------------------------------------------------------------
+
+"""
+
+# %%
+# Step 1: Importing the Libraries
+# ----------------------------------
+import typing
+
+import flytekit
+import pandas as pd
+from flytekit import Resources, dynamic, task, workflow
+from flytekit.types.file import FlyteFile
+
+try:
+    from .house_price_predictor import (
+        generate_and_split_data,
+        fit,
+        predict,
+    )
+except ImportError:
+    from house_price_predictor import (
+        generate_and_split_data,
+        fit,
+        predict,
+    )
+
+# %%
+# Step 2: Initializing the Variables
+# ----------------------------------
+NUM_HOUSES_PER_LOCATION = 1000
+COLUMNS = [
+    "PRICE",
+    "YEAR_BUILT",
+    "SQUARE_FEET",
+    "NUM_BEDROOMS",
+    "NUM_BATHROOMS",
+    "LOT_ACRES",
+    "GARAGE_SPACES",
+]
+LOCATIONS = [
+    "NewYork_NY",
+    "LosAngeles_CA",
+    "Chicago_IL",
+    "Houston_TX",
+    "Dallas_TX",
+    "Phoenix_AZ",
+    "Philadelphia_PA",
+    "SanAntonio_TX",
+    "SanDiego_CA",
+    "SanFrancisco_CA",
+]
+
+# %%
+# Step 3: Task -- Generating & Splitting the Data for Multiple Regions
+# --------------------------------------------------------------------
+#
+# Train, validation, and test datasets are lists of DataFrames.
+@dynamic(cache=True, cache_version="0.1", limits=Resources(mem="600Mi"))
+def generate_and_split_data_multiloc(
+    locations: typing.List[str],
+    number_of_houses_per_location: int,
+    seed: int,
+) -> (typing.List[pd.DataFrame], typing.List[pd.DataFrame], typing.List[pd.DataFrame]):
+    train_sets = []
+    val_sets = []
+    test_sets = []
+    for loc in locations:
+        _train, _val, _test = generate_and_split_data(
+            loc=loc, number_of_houses=number_of_houses_per_location, seed=seed
```

Comment:
Let's remove loc from the input list of the `generate_and_split_data` task.  It's not used.

Reply:
Done.